### DISCUSSION TITLE
prerequisites

### DISCUSSION THREAD

**Discussion by zehypnos** [ksheumaker/homeassistant-apsystems_ecur#242] [Q&A]
Good morning ! first of all thank you for this development and this time spent for this integration.
I want to add this integration but from the start it doesn't work.

At the command: `[core-ssh .storage]$ nc -v 192.168.1.4 8899` I have no response.
However, I can ping or nslookup my ECU-B.
Do you think port 8899 might be blocked somewhere?

Although I have searched and tested in all directions, for the moment I am making no progress...
I am therefore counting on your help.

Have a good day .

**Comment by HAEdwin** [comment]
If other ESP network interfaces exist, please make sure you have the right node (ECU-B). The NC command should then provide a response, after all, the ECU-B is also installed via the setup of the ECU manager app, I assume? Try ping -t while turning off the ECU to make absolutely sure you have the correct node.

**Reply by zehypnos** [reply]
Good morning ! yes I checked that it was the correct IP for the ECU. And yes I already use it in the ECU manager application.

**Comment by zehypnos** [comment]
![image](https://github.com/ksheumaker/homeassistant-apsystems_ecur/assets/78178877/c5f93185-101f-4d88-a1f5-7967930c112f)

**Reply by HAEdwin** [reply]
Have you allready rebooted the ECU? What firmware?

**Reply by zehypnos** [reply]
Hello, yes i'm rebooted the ECU.  
How can I find out the firmware version ?

**Reply by HAEdwin** [reply]
Have you allready tried to install the integration? Firmware can be found in the ECU manager app. The port is only available in WiFi mode, not ethernet.

**Reply by zehypnos** [reply]
Hello, in ECU manager app, I can't find the firmware, I only have this information:

![image](https://github.com/ksheumaker/homeassistant-apsystems_ecur/assets/78178877/ce0f465f-25ac-449a-9495-33b75b0ed012)
or
![image](https://github.com/ksheumaker/homeassistant-apsystems_ecur/assets/78178877/7df3987b-10a7-4066-a260-05a01d569213)

Yes, the ECU is on wifi and I have already tried installing the integration.

![image](https://github.com/ksheumaker/homeassistant-apsystems_ecur/assets/78178877/b1561c17-2893-43b4-89ed-6a0f1e923ac2)

**Reply by HAEdwin** [reply]
Might there be something blocking 8899 over the network? I see you use the EMA app but you should use the ECU app. If you are able to configure the ECU using the temporary hotspot, you should also be able to do this without using the hotspot (hotspot will go offline after a while). It is strange though, there are ECU-B users out there who do not have any difficulties.

**Comment by zehypnos** [comment]
I can't find this ECU application? I didn't know there were several. I'm looking on my network for potential blocking but I can't find anything. From my PC, if I `telnet 192.168.1.4 8899` it replies:  _Unable to open a connection to the host, on port 8883: Failure during connection_
it's a way to see if the port is listening but we can see that it's not.

**Reply by HAEdwin** [reply]
Indeed the ECU app has left the store. In the EMA manager > Workspace you should be able to see the firmware version.
![image](https://github.com/ksheumaker/homeassistant-apsystems_ecur/assets/82239374/1ae24ac9-c1c4-4eb3-91d4-c1c2d91897e0)

Have you allready tried a port scan? What are the LAN settings in the EMA manager app (Workspace > ECU Metwork Settings > LAN Settings)?

**Reply by zehypnos** [reply]
unfortunately in ema ECU app there is no information about the network config, just the possibility of an option to replace this ECU.

![image](https://github.com/ksheumaker/homeassistant-apsystems_ecur/assets/78178877/d62df76d-fb36-4505-947c-19481c2493a9)

![image](https://github.com/ksheumaker/homeassistant-apsystems_ecur/assets/78178877/a9c1fc6b-83c9-488c-9926-7f1489ebe9ed)

![image](https://github.com/ksheumaker/homeassistant-apsystems_ecur/assets/78178877/1c8162d9-225a-4e2a-81b0-41fed2090d95)

I went through all the menus to no avail.

**Comment by zehypnos** [comment]
I did a port scan on the ECU address
![image](https://github.com/ksheumaker/homeassistant-apsystems_ecur/assets/78178877/5e0fc645-0ab9-4f12-a147-581eae968d28)

**Reply by HAEdwin** [reply]
These ports are not typically linked to APSystems. I would expect the hostname to start with ESP_ followed by the last 6 digits of the MAC address. Are you 100% sure you have the correct node (you would not be the first to have picked the wrong Espressif Inc. node). The screenprint I posted is the EMA Manager app.

As a last ditch effort, try packet sender:
![image](https://github.com/ksheumaker/homeassistant-apsystems_ecur/assets/82239374/daa41c47-aa3f-46a0-85ae-70cc9d5692a1)


If this doesn't work either the port is closed by the firmware or the node is wrong/blocked on 8899

**Reply by zehypnos** [reply]
Good morning,

You were right, never be sure of yourself. I checked again if I was on the right IP, and yes from the start I was wrong, the address is not the right one.
The command `nc -v 192.168.1.4 8899` responds `open`. But the command `APS1100160001END` does not respond anything.
I still tried the integration and everything went well !

I really want to thank you for taking the time to help me and especially for your patience. I'm obviously a little ashamed to be so stupidly mistaken but for me I had checked by cutting but I had to get tangled up in my IP addresses which are starting to be numerous.

Bonne journée.

**Reply by HAEdwin** [reply]
No problem, I appreciate your honesty. Glad it's been resolved.
